### PR TITLE
Fixes tile palette removing, stuck fire, Locks reagent dict in metadatalayer, Moves discord link to config

### LINKED ISSUE #6823: Tile Palette Erasure Bug in Unity Editor
### Environment
- Game Version: Doesn't Matter Editor Bug
- OS Version: Windows 10

### Detailed description of bug
Can't erase tiles that you place using the tile palette because of a NRE.
image

### Steps to reproduce

1. Open Tile palette in Unity Editor
2. Attempt to erase a tile on any matrix in a scene

### PR DESCRIPTION
**Pull request by ThatDan123**
-Fixes tile palette removing
Fixes #6823

-Checks for null to stop mob NRE error, as it could be null at that point

-Locks reagent dict in metadatalayer

-Hopefully fixes stuck fire

-Error log is now on for all categories

-Moves discord link to config as it doesnt need to be in separate file anymore

### CODE REVIEW

File: UnityProject/Assets/Scripts/Messages/Server/TabUpdateMessage.cs [hunk 1]
```diff
@@ -120,7 +120,7 @@ public static void SendToPeepers(GameObject provider, NetTabType type, TabAction
 					var instance = NetworkTabManager.Instance.Get(provider, type);
 					if (instance == null)
 					{
-						Logger.LogError($"Couldn't find NetTab to send! " +
+						Logger.LogError($"Couldn't find NetTab to send for {provider.ExpensiveName()} " +
 								$"Does the tab prefab match the type '{type}'? Expecting prefab name Tab'{type}' in a specific folder.");
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Messages/Server/TabUpdateMessage.cs]:
```suggestion
								$"Does the tab prefab match the type '{type}'? Expecting prefab name 'Tab{type}' in a specific folder.");
```
Just fixing my mistake...

Edit: dang it, didn't want to comment this until just before I merged, but the other comment submitted this one as well ; )

**Comment by Bod9001** [comment]
looks good

**Comment by Aranclanos** [comment]
I'm really worried about all the null checks added at seemingly random areas

**Review by NoooneyDude — COMMENTED** [review]
> I'm really worried about all the null checks added at seemingly random areas

Might be a good idea to comment all instances of null checks if not immediately obvious.

**Comment by ThatDan123** [comment]
> I'm really worried about all the null checks added at seemingly random areas

Which ones are you talking about?

**Comment by Aranclanos** [comment]
TileChangeNewPlayer, Canister, PlayerMove, MachineFrame and GhostRoleManager

**Comment by ThatDan123** [comment]
TileChangeNewPlayer: Added a log to record it so we still catch the error

`System.NullReferenceException: Object reference not set to an instance of an object
  at Messages.Client.NewPlayer.TileChangeNewPlayer.Process (Messages.Client.NewPlayer.TileChangeNewPlayer+NetMessage msg) [0x0000d] in /root/UnityProject/Assets/Scripts/Messages/Client/NewPlayer/TileChangeNewPlayer.cs:15 
  at Messages.GameMessageBase`1[T].Process (Mirror.NetworkConnection sentBy, T msg) [0x00000] in /root/UnityProject/Assets/Scripts/Messages/GameMessageBase.cs:28 
UnityEngine.Debug:LogError(Object)
Messages.GameMessageBase`1:Process(NetworkConnection, NetMessage) (at /root/UnityProject/Assets/Scripts/Messages/GameMessageBase.cs:32)
Messages.Client.ClientMessage`1:Process(NetworkConnection, NetMessage) (at /root/UnityProject/Assets/Scripts/Messages/Client/ClientMessage.cs:15)
Messages.GameMessageBase`1:PreProcess(NetworkConnection, NetMessage) (at /root/UnityProject/Assets/Scripts/Messages/GameMessageBase.cs:17)
Mirror.<>c__DisplayClass5_0`2:<WrapHandler>b__0(NetworkConnection, NetworkReader, Int32) (at /root/UnityProject/Assets/Mirror/Runtime/MessagePacking.cs:118)
Mirror.NetworkConnection:UnpackAndInvoke(NetworkReader, Int32) (at /root/UnityProject/Assets/Mirror/Runtime/NetworkConnection.cs:194)
Mirror.NetworkConnection:TransportReceive(ArraySegment`1, Int32) (at /root/UnityProject/Assets/Mirror/Runtime/NetworkConnection.cs:232)
Mirror.NetworkServer:OnDataReceived(Int32, ArraySegment`1, Int32) (at /root/UnityProject/Assets/Mirror/Runtime/NetworkServer.cs:598)
IgnoranceTransport.Ignorance:ProcessServerPackets() (at /root/UnityProject/Assets/Mirror/Runtime/Transport/Ignorance/Ignorance.cs:420)
IgnoranceTransport.Ignorance:ProcessAndExecuteAllPackets() (at /root/UnityProject/Assets/Mirror/Runtime/Transport/Ignorance/Ignorance.cs:581)
IgnoranceTransport.Ignorance:Update() (at /root/UnityProject/Assets/Mirror/Runtime/Transport/Ignorance/Ignorance.cs:570)`

**Comment by ThatDan123** [comment]
Canister was caused by having the tank open and removing the inserted canister. The update loop called the function to add more gas to the tank but doesnt have a tank anymore

**Comment by ThatDan123** [comment]
PlayerMove was the connection is null when the owner player is disconnected

**Comment by ThatDan123** [comment]
Machine frame one seems like theres a prefab which has lost a reference, so the log was added to catch it

**Comment by ThatDan123** [comment]
Ghost roll stuff i went through with noooney

**Pull request merged**